warehouse/internal/repo/staging.go

```diff
@@ -342,23 +342,10 @@ func (repo *StagingFiles) CountPendingForDestination(ctx context.Context, destin
 }
 
 func (repo *StagingFiles) countPending(ctx context.Context, query string, value interface{}) (int64, error) {
-	var id sql.NullInt64
+	var count int64
 	err := repo.db.QueryRowContext(ctx,
-		`SELECT MAX(end_staging_file_id) FROM `+uploadsTableName+` WHERE `+query,
+		`SELECT COUNT(*) FROM `+stagingTableName+` WHERE `+query+` AND id > (SELECT COALESCE(MAX(end_staging_file_id), 0) FROM `+uploadsTableName+` WHERE `+query+`)`,
```

Comment:
Is `end_staging_file_id` indexed? Should we expect considerable performance degradation with the new query?

Reply:
Currently, we don't have indexes present.